browser/ui/views/frame/brave_browser_view.cc

```diff
@@ -347,6 +358,56 @@ void BraveBrowserView::OnTabStripModelChanged(
   }
 }
 
+views::CloseRequestResult BraveBrowserView::OnWindowCloseRequested() {
+  if (GetBraveBrowser()->ShouldAskForBrowserClosingBeforeHandlers()) {
+    WindowClosingConfirmDialogView::Show(
```

Comment:
can you please check if everything is fine when you have 2 profiles, have windows opened (with 2 tabs) for both profiles and delete a profile from profile manage window. This should close the window for a second profile.

Reply:
Thanks! I didn't know this scenario.
This PR doesn't work with profile delete scenario! 👍🏼

Reply:
I mean it's fine if we don't show the dialog at this exact case, just make sure we don't crash or do something weird there.

Reply:
Yeah, dialog shoudn't be launched but dialog is launched :)
When I select `cancel` window closing via dialog, profile is deleted but window is still living.
In this case, return value of this method seems ignored.